x86_64-mobilecoin-none-sgx.json

```diff
@@ -0,0 +1,67 @@
+{
+  "arch": "x86_64",
+  "cpu": "x86-64",
+  "crt-static-respected": true,
+  "data-layout": "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128",
+  "dynamic-linking": true,
+  "env": "sgx",
+  "executables": true,
+  "features": "+rdrnd,+rdseed,+lvi-cfi,+lvi-load-hardening",
+  "has-rpath": true,
+  "has-thread-local": true,
+  "llvm-args": [
+    "--x86-experimental-lvi-inline-asm-hardening"
+  ],
+  "llvm-target": "x86_64-unknown-linux-gnu",
+  "max-atomic-width": 64,
+  "os": "none",
+  "position-independent-executables": true,
+  "pre-link-args": {
+    "gcc": [
+      "-m64"
+    ],
+    "ld.lld": [
+      "-e",
+      "elf_entry",
+      "-Bstatic",
+      "--gc-sections",
+      "-z",
+      "text",
+      "-z",
+      "norelro",
+      "--no-undefined",
+      "--error-unresolved-symbols",
+      "--no-undefined-version",
+      "-Bsymbolic",
+      "--export-dynamic",
+      "-u",
+      "__rust_abort",
+      "-u",
+      "__rust_c_alloc",
+      "-u",
+      "__rust_c_dealloc",
+      "-u",
+      "__rust_print_err",
+      "-u",
+      "__rust_rwlock_rdlock",
+      "-u",
+      "__rust_rwlock_unlock",
+      "-u",
+      "__rust_rwlock_wrlock"
+    ]
+  },
```

Comment:
This is all fortanix-specifc, and shouldn't be here
```suggestion
```

Reply:
The actual linker args we use with an enclave are at https://github.com/mobilecoinfoundation/mobilecoin/blob/878eceb052d992c09d01b12db2e03d153b887d8e/util/build/enclave/src/lib.rs#L739

Reply:
Took a bit to get through my head.
Things like [cc](https://docs.rs/cc/latest/cc/) and [bindgen](https://docs.rs/bindgen/latest/bindgen/) currently don't have a clean way to get to the linker args.  However if the SGX `build.rs` is compiling rust files, then if done right, we should be able to move the link args to the .json file and the `build.rs` can avoid _manually_ compiling the rs files.  The link args should be passed correctly from cargo to rustc.

Reply:
This should be addressed now, with the exception that the `_sim` libraries are still looked up via a build script.  The future plan is to create a sim specific spec file